lib/client/mcp/claude/config_test.go

```diff
@@ -0,0 +1,146 @@
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package claude
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/stretchr/testify/require"
+)
+
+const sampleConfigJSON = `{
+  "someUnknownField": "someUnknownValue",
+  "mcpServers": {
+    "Puppeteer": {
+      "command": "npx",
+      "args": [
+        "-y",
+        "@modelcontextprotocol/server-puppeteer"
+      ],
+      "someUnknownField": "someUnknownValue"
+    },
+    "teleport-my-mcp": {
+      "command": "tsh",
+      "args": [
+        "mcp",
+        "connect",
+        "my-mcp"
+      ],
+      "env": {
+        "TELEPORT_HOME": "/tsh-home/"
+      }
+    }
+  }
+}
+`
+
+var sampleConfig = Config{
+	MCPServers: map[string]MCPServer{
+		"Puppeteer": {
+			Command: "npx",
+			Args:    []string{"-y", "@modelcontextprotocol/server-puppeteer"},
+		},
+		"teleport-my-mcp": {
+			Command: "tsh",
+			Args:    []string{"mcp", "connect", "my-mcp"},
+			Envs: map[string]string{
+				"TELEPORT_HOME": "/tsh-home/",
+			},
+		},
+	},
```

Comment:
I think it would have helped me understand if `AllFields` and `"someUnknownField"` had been kept in here.